core/src/main/java/org/opensearch/sql/executor/QueryService.java

```diff
@@ -250,8 +250,14 @@ public PhysicalPlan plan(LogicalPlan plan) {
     return planner.plan(plan);
   }
 
-  public RelNode optimize(RelNode plan) {
-    return planner.customOptimize(plan);
+  /**
+   * Try to optimize the plan by appending a limit operator for QUERY_SIZE_LIMIT Don't add for
+   * `EXPLAIN` to avoid changing its output plan.
+   */
+  public RelNode optimize(RelNode plan, CalcitePlanContext context, boolean isExplain) {
+    if (isExplain) return plan;
+    context.relBuilder.limit(0, context.querySizeLimit);
```

Comment:
can you add a case in `ExplainIT`?

Reply:
Add an IT to verify querySizeLimit pushdown.
Q, should we revert code of https://github.com/opensearch-project/sql/pull/3880?

Reply:
We can reuse [LogicalSystemLimit](https://github.com/opensearch-project/sql/pull/3749) in explain. The physical plan is no diff. Then we can verify the querySizeLimit pushdown in explain IT.